modules/pcomp/pcomp_doc.rst

```diff
@@ -0,0 +1,238 @@
+PCOMP - Position Compare [x4]
+=============================
+
+The position compare block takes a position input and allows a regular number
+of threshold comparisons to take place on a position input. The normal order
+of operations is something like this:
+
+* If PRE_START > 0 then wait until position has passed START - PRE_START
+* If START > 0 then wait until position has passed START and set OUT=1
+* Wait until position has passed START + WIDTH and set OUT=0
+* Wait until position has passed START + STEP and set OUT=1
+* Wait until position has passed START + STEP + WIDTH and set OUT=0
+* Continue until PULSES have been produced
+
+It can be used to generate a position based pulse train against an input encoder
+or analogue system, or to work as repeating comparator.
+
+
+Fields
+------
+.. block_fields:: modules/pcomp/pcomp.block.ini
+
+Position compare is directional
+-------------------------------
+
+A typical example would setup the parameters, enable the block, then start
+moving a motor to trigger a series of pulses:
+
+.. timing_plot::
+   :path: modules/pcomp/pcomp.timing.ini
+   :section: 3 Pulses in a +ve direction
+
+But if we get the direction wrong, we won't get the first pulse until we cross
+START in the correct direction:
+
+.. timing_plot::
+   :path: modules/pcomp/pcomp.timing.ini
+   :section: Enabled while crossing in wrong direction
+
+Moving in a negative direction works in a similar way. Note that WIDTH and
+PULSE still have positive values:
+
+.. timing_plot::
+   :path: modules/pcomp/pcomp.timing.ini
+   :section: 2 Pulses in a -ve direction
+
+Internal statemachine
+---------------------
+
+The Block has an internal statemachine that is exposed as a parameter, allowing
+the user to see what the Block is currently doing:
+
+.. digraph:: pcomp_sm
```

Comment:
If you see another block with STATE can you make sure there is a matching state diagram please?